cluster/discovery/discovery.go

```diff
@@ -0,0 +1,33 @@
+package discovery
+
+// NodeEntry is used to discovery nodes in the cluster
+type NodeEntry struct {
+	Id            string
+	Ip            string
+	GossipVersion string
```

Comment:
Currently we check gossip version before we pass in the list of Ips to gossip.Start.